src/classes/UTIL_OrgTelemetry_SVC.cls

```diff
@@ -195,7 +212,79 @@ public without sharing class UTIL_OrgTelemetry_SVC {
                 languages.add(u.LanguageLocaleKey);
             }
         }
-        System.FeatureManagement.setPackageIntegerValue(TelemetryParameterName.Org_CountNonEnglishLanguages.name(), languages.size());
+        featureManager.setPackageIntegerValue(TelemetryParameterName.Org_CountNonEnglishLanguages.name(), languages.size());
+    }
+
+    /**
+     * @description Captures the longest running batch time in minutes of the last 7 days.
+     */
+    private void handleLongestRunningBatchLast7Days() {
+        Integer longestJobLength = 0;
+
+        for (AsyncApexJob job : [
+            SELECT CreatedDate, CompletedDate
+            FROM AsyncApexJob
+            WHERE CreatedDate = LAST_N_DAYS:7
+                AND JobType = 'BatchApex'
+                AND Status IN ('Aborted', 'Completed', 'Failed')
+                AND ApexClass.NamespacePrefix IN ('npsp', 'npe01', 'npo02', 'npe03', 'npe4', 'npe5')
+        ]) {
+
+            Integer jobLength = getMinutesBetween(job.CreatedDate, job.CompletedDate);
+            longestJobLength = jobLength > longestJobLength ? jobLength : longestJobLength;
+
+        }
+
+        featureManager.setPackageIntegerValue(TelemetryParameterName.Org_LongestRunningBatchLast7Days.name(), longestJobLength);
+    }
+
+    /**
+     * @description Calculates the number of minutes between two date times
+     */
+    private Integer getMinutesBetween(DateTime startDateTime, DateTime endDateTime) {
```

Comment:
Not sure this is correct. I tried the following code. The result should be 11 hours (660 minutes), however the method returns 2100 minutes (35 hours).
```
DateTime startDate = DateTime.newInstanceGMT(2019, 03, 01, 23, 00,00);
DateTime endDate = DateTime.newInstanceGMT(2019, 03, 02, 10, 00,00);
System.debug(startDate + ' -> ' + endDate);
System.debug(getMinutesBetween(startDate, endDate));
```

Reply:
Ah for some reason I was removing a day from the start date in the method, probably from when I was originally testing the logic. Good catch! I'll add a unit test, date/time calculations are confusing enough to warrant one even if it is a private method